In [27]:
import pandas as pd

# Step 1: Load the Value and Volume Dataset
my_data = "Data Test.xlsx"
value_df = pd.read_excel(my_data, sheet_name="Value", header=11)
volume_df = pd.read_excel(my_data, sheet_name="Volume", header=11)

# show few rows
value_df.head()
volume_df.head()


Unnamed: 0  Unnamed: 1 Price Sector Reportable Owner       Brand House  \
0         NaN         NaN      Economy              BAT         Pall Mall   
1         NaN         NaN      Economy              BAT          Rothmans   
2         NaN         NaN      Economy             ITUK               JPS   
3         NaN         NaN      Economy             ITUK  Lambert & Butler   
4         NaN         NaN      Economy             ITUK          Player's   

   w/e 04/10/2020  w/e 11/10/2020  w/e 18/10/2020  w/e 25/10/2020  \
0     4143.934021     8398.187367      362.963860     1211.434820   
1     7842.824386     7634.001584     5931.576242     7311.134212   
2     6155.589941     2788.480471     8967.033939     7704.356320   
3     7989.661541     9448.859414     8441.028687     5031.936885   
4     7656.676748     9360.070640     4951.920646     9395.296091   

   w/e 01/11/2020  ...  w/e 19/01/2025  w/e 26/01/2025  w/e 02/02/2025  \
0     8251.210643  ...     7997.348378     6917.606883     6033.091229   
1     5936.685903  ...     4814.565415     2770.990240     9289.269846   
2     9483.384989  ...     5470.431170     5900.342122     7655.683206   
3     6149.225638  ...     3834.727972     6246.944214     8510.271928   
4     5032.660859  ...     7693.330674     8046.840831     8258.273476   

   w/e 09/02/2025  w/e 16/02/2025  w/e 23/02/2025  w/e 02/03/2025  \
0     9901.514201     2341.929237      851.308635     1632.851576   
1     7375.029871     5802.894218      421.653166     8557.013335   
2     4691.870134     8941.903926     4134.719276     5654.416316   
3     8543.052036      195.307934      949.187339     8565.401761   
4     6138.467134     2596.772975     2729.335533     7893.594663   

   w/e 09/03/2025  w/e 16/03/2025  w/e 23/03/2025  
0     4837.892529     1380.733131     9092.091805  
1     2903.867314     6781.692145     9650.355703  
2     5942.757288     5877.299934     7368.217115  
3     4770.939300     9761.625931     2258.235194  
4     9781.622692      990.542782     3855.249908  

[5 rows x 239 columns]

In [29]:
# Keep only relevant columns
value_cols = [col for col in value_df.columns if isinstance(col, str) and col.startswith("w/e")]
volume_cols = [col for col in volume_df.columns if isinstance(col, str) and col.startswith("w/e")]

value_df = value_df[["Price Sector", "Reportable Owner", "Brand House"] + value_cols]
volume_df = volume_df[["Price Sector", "Reportable Owner", "Brand House"] + volume_cols]

value_df.head()


Price Sector Reportable Owner       Brand House  w/e 04/10/2020  \
0      Economy              BAT         Pall Mall     3665.618289   
1      Economy              BAT          Rothmans      561.383570   
2      Economy             ITUK               JPS      453.451338   
3      Economy             ITUK  Lambert & Butler     3409.473332   
4      Economy             ITUK          Player's     4048.641554   

   w/e 11/10/2020  w/e 18/10/2020  w/e 25/10/2020  w/e 01/11/2020  \
0     2172.201949     1171.715922     7857.273171     7528.706700   
1     8713.255516     5074.939497     8792.736587     4788.023222   
2     3293.169146     1744.711615     3136.655381     5258.759882   
3     1444.240027     6594.074798     1550.628133     4401.356623   
4     4211.961108     2221.901403     4454.086423     2620.333824   

   w/e 08/11/2020  w/e 15/11/2020  ...  w/e 19/01/2025  w/e 26/01/2025  \
0     2075.037029     4636.963146  ...     7756.351578     3280.731901   
1     2918.215462      398.552955  ...     1397.829368     6323.083353   
2     6180.779319     8788.305471  ...     8638.986706      697.324122   
3     8568.767891     1459.817046  ...      668.059250        5.232344   
4     5849.587212     7781.720135  ...     6635.080032     4684.846233   

   w/e 02/02/2025  w/e 09/02/2025  w/e 16/02/2025  w/e 23/02/2025  \
0     1147.694494     6244.074922     1397.590611     4035.041068   
1     6137.819981     1736.129950     5579.069294      258.552528   
2     7705.210566     2052.634317     6621.006239     6238.445136   
3     6255.575643     1158.046116     7259.476575     1767.423131   
4     8373.961562       41.923165     9733.976606     3300.519994   

   w/e 02/03/2025  w/e 09/03/2025  w/e 16/03/2025  w/e 23/03/2025  
0     5280.808351     8625.630285     4447.797944      674.448281  
1      448.083924     4399.173558     5555.501250     4032.278734  
2     8789.066553     7889.363412     2128.827701     2068.969338  
3     3970.681116     1347.427859     2324.590337     8600.946996  
4     4111.638114      838.058816     9391.814382     3040.768702  

[5 rows x 237 columns]

In [30]:
# Reshape value and volume data from wide to long format
value_long = pd.melt(
    value_df,
    id_vars=["Price Sector", "Reportable Owner", "Brand House"],
    var_name="Date",
    value_name="Value"
)

volume_long = pd.melt(
    volume_df,
    id_vars=["Price Sector", "Reportable Owner", "Brand House"],
    var_name="Date",
    value_name="Volume"
)

value_long.head()




Price Sector Reportable Owner       Brand House            Date        Value
0      Economy              BAT         Pall Mall  w/e 04/10/2020  3665.618289
1      Economy              BAT          Rothmans  w/e 04/10/2020   561.383570
2      Economy             ITUK               JPS  w/e 04/10/2020   453.451338
3      Economy             ITUK  Lambert & Butler  w/e 04/10/2020  3409.473332
4      Economy             ITUK          Player's  w/e 04/10/2020  4048.641554

In [31]:
# Remove 'w/e ' prefix and convert to datetime
value_long["Date"] = pd.to_datetime(value_long["Date"].str.replace("w/e ", ""), dayfirst=True)
volume_long["Date"] = pd.to_datetime(volume_long["Date"].str.replace("w/e ", ""), dayfirst=True)

value_long["Date"].head()



0   2020-10-04
1   2020-10-04
2   2020-10-04
3   2020-10-04
4   2020-10-04
Name: Date, dtype: datetime64[ns]

In [32]:
# Merge value and volume data
merged_data = pd.merge(
    value_long,
    volume_long,
    on=["Price Sector", "Reportable Owner", "Brand House", "Date"],
    how="inner"
)

merged_data.head()


Price Sector Reportable Owner       Brand House       Date        Value  \
0      Economy              BAT         Pall Mall 2020-10-04  3665.618289   
1      Economy              BAT          Rothmans 2020-10-04   561.383570   
2      Economy             ITUK               JPS 2020-10-04   453.451338   
3      Economy             ITUK  Lambert & Butler 2020-10-04  3409.473332   
4      Economy             ITUK          Player's 2020-10-04  4048.641554   

        Volume  
0  4143.934021  
1  7842.824386  
2  6155.589941  
3  7989.661541  
4  7656.676748

In [34]:
# Create Holding_Company column by replacing ITUK with IMB
merged_data["Holding_Company"] = merged_data["Reportable Owner"].replace("ITUK", "IMB")

# check column
merged_data[["Reportable Owner", "Holding_Company"]].drop_duplicates().head()


Reportable Owner  Holding_Company
0              BAT              BAT
2             ITUK              IMB
5  ITUK (Pr Label)  ITUK (Pr Label)
6              JTI              JTI
8            Other            Other

In [36]:
# Create PS_HC_BH_Group column
merged_data["PS_HC_BH_Group"] = (
    merged_data["Price Sector"] + "_" +
    merged_data["Holding_Company"] + "_" +
    merged_data["Brand House"]
)

# check column
merged_data[["PS_HC_BH_Group"]].head()


PS_HC_BH_Group
0         Economy_BAT_Pall Mall
1          Economy_BAT_Rothmans
2               Economy_IMB_JPS
3  Economy_IMB_Lambert & Butler
4          Economy_IMB_Player's

In [37]:
# Calculate total value and volume per Date
merged_data["Value_T"] = merged_data.groupby("Date")["Value"].transform("sum")
merged_data["Volume_T"] = merged_data.groupby("Date")["Volume"].transform("sum")


In [48]:
# Calculate share columns
merged_data["Value_Share"] = merged_data["Value"] / merged_data["Value_T"]
merged_data["Volume_Share"] = merged_data["Volume"] / merged_data["Volume_T"]

# check columns
merged_data[[
    "Date", "Price Sector", "Holding_Company", "Brand House","PS_HC_BH_Group",
    "Value", "Volume", "Value_T", "Volume_T", "Value_Share", "Volume_Share"
]].head()


Date Price Sector Holding_Company       Brand House  \
0 2020-10-04      Economy             BAT         Pall Mall   
1 2020-10-04      Economy             BAT          Rothmans   
2 2020-10-04      Economy             IMB               JPS   
3 2020-10-04      Economy             IMB  Lambert & Butler   
4 2020-10-04      Economy             IMB          Player's   

                 PS_HC_BH_Group        Value       Volume        Value_T  \
0         Economy_BAT_Pall Mall  3665.618289  4143.934021  298503.214225   
1          Economy_BAT_Rothmans   561.383570  7842.824386  298503.214225   
2               Economy_IMB_JPS   453.451338  6155.589941  298503.214225   
3  Economy_IMB_Lambert & Butler  3409.473332  7989.661541  298503.214225   
4          Economy_IMB_Player's  4048.641554  7656.676748  298503.214225   

        Volume_T  Value_Share  Volume_Share  
0  316264.844844     0.012280      0.013103  
1  316264.844844     0.001881      0.024798  
2  316264.844844     0.001519      0.019463  
3  316264.844844     0.011422      0.025263  
4  316264.844844     0.013563      0.024210

In [45]:
# Step 10: Final answer filtering
final_output = merged_data[
    (merged_data["Price Sector"] == "Premium") &
    (merged_data["Holding_Company"] == "IMB") &
    (merged_data["Brand House"] == "JPS") &
    (merged_data["Date"] == pd.to_datetime("2024-04-21"))
][["Date", "Value_Share", "Volume_Share"]]

# Show the result
final_output

Date  Value_Share  Volume_Share
11498 2024-04-21     0.030218      0.024012

What is the value share and volume share for JPS, under IMB, in the Premium price sector on 21/04/2024?

from the result 

Value Share = 3.02%

Volume Share = 2.40%



In [47]:
#save result
final_output.to_csv("jps_share_2024_04_21.csv", index=False)
